File: backend/python/airdrop_usdtg.py
import os
import time
from dotenv import load_dotenv
from tronpy import Tron
from tronpy.exceptions import TransactionError

load_dotenv(dotenv_path=os.path.join(os.path.dirname(__file__), '../../justland/.env'))

PRIVATE_KEY = os.getenv("PRIVATE_KEY")
USDTG_CONTRACT = os.getenv("USDTG_CONTRACT")
OWNER_ADDRESS = os.getenv("OWNER_ADDRESS")

client = Tron()
wallet = client.generate_address()
account = client.get_account(OWNER_ADDRESS)

contract = client.get_contract(USDTG_CONTRACT)

def load_recipients(filepath):
    with open(filepath, 'r') as file:
        return [line.strip() for line in file.readlines() if line.strip().startswith('T')]

def send_airdrop(recipient, amount_sun):
    try:
        txn = (
            contract.functions.transfer(recipient, amount_sun)
            .with_owner(OWNER_ADDRESS)
            .build()
            .sign(PRIVATE_KEY)
            .broadcast()
        )
        txn.wait()
        print(f"✅ Sent to {recipient} | TxID: {txn.txid}")
    except TransactionError as e:
        print(f"❌ Failed to send to {recipient}: {e}")
    except Exception as ex:
        print(f"❌ General error: {ex}")

def main():
    print("🚀 USDTg Airdrop Başlatıldı...")

    recipients = load_recipients('recipients.txt')
    amount = 100 * 1_000_000  # 6 decimals for USDTg

    for idx, recipient in enumerate(recipients, 1):
        print(f"📤 {idx}/{len(recipients)}: {recipient}")
        send_airdrop(recipient, amount)
        time.sleep(0.25)

    print("🎉 Airdrop tamamlandı!")

if __name__ == "__main__":
    main()

File: backend/python/airdropUSDTg.js
import { TronWeb } from 'tronweb';
import fs from 'fs/promises';
import dotenv from 'dotenv';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
dotenv.config({ path: path.resolve(__dirname, '../../justland/.env') });

// === ENV'DEN YÜKLENEN DEĞERLER ===
const { PRIVATE_KEY, USDTG_CONTRACT, OWNER_ADDRESS } = process.env;

const tronWeb = new TronWeb({
  fullHost: 'https://api.trongrid.io',
  privateKey: PRIVATE_KEY,
});

async function loadRecipients(filePath) {
  const data = await fs.readFile(filePath, 'utf8');
  return data
    .split('\n')
    .map((line) => line.trim())
    .filter((addr) => tronWeb.isAddress(addr));
}

async function sendAirdrop(recipient, amountSun, usdtg) {
  try {
    const tx = await usdtg.methods
      .transfer(recipient, amountSun)
      .send({ shouldPollResponse: true });
    console.log(`✅ Sent to ${recipient} | TxID: ${tx}`);
  } catch (err) {
    console.error(`❌ Failed to send to ${recipient}:`, err.message || err);
  }
}

async function main() {
  console.log('🚀 USDTg Airdrop Başlatıldı...');

  const usdtg = await tronWeb.contract().at(tronWeb.address.toHex(USDTG_CONTRACT));
  const recipients = await loadRecipients(path.resolve(__dirname, 'recipients.txt'));

  const amount = tronWeb.toSun(100); // 100 USDTg
  for (let i = 0; i < recipients.length; i++) {
    const addr = recipients[i];
    console.log(`📤 ${i + 1}/${recipients.length}: ${addr}`);
    await sendAirdrop(addr, amount, usdtg);
    await new Promise((r) => setTimeout(r, 250)); // küçük delay ekle
  }

  console.log('🎉 Airdrop tamamlandı!');
}

main();